R/particle_filter_state_nested.R

```diff
@@ -208,18 +233,19 @@ particle_filter_state_nested <- R6::R6Class(
         if (is.null(compare)) {
           log_weights <- model$compare_data()
         } else {
-          ok <- !is.null(compare(array(state[, , 1],
-                                       c(nrow(state), ncol(state))),
-                                 data_split[[t]][[1]], pars))
+          ok <- !is.null(compare(
```

Comment:
Why are we running compare() twice here?

Reply:
No idea....

Reply:
Oh! Because of the `vapply` https://github.com/mrc-ide/mcstate/blob/e4798e9907f41f46bda0cf24459b316b385232cc/R/particle_filter_state_nested.R#L243-L248 . I guess in testing I found it could return NULL for some `t` but not all, and running the `vapply` threw errors when this was the case

Reply:
in which case, run the main loop in an lapply, then a second vapply to collect together